ndk/src/native_window.rs

```diff
@@ -126,4 +126,62 @@ impl NativeWindow {
     pub unsafe fn to_surface(&self, env: *mut JNIEnv) -> jobject {
         ffi::ANativeWindow_toSurface(env, self.ptr().as_ptr())
     }
+
+    /// Lock the window's next drawing surface for writing.
+    ///
+    /// Optionally pass the region you intend to draw into `dirty_bounds`.  When this function
+    /// returns it is updated (commonly enlarged) with the actual area the caller needs to redraw.
+    pub fn lock(
+        &self,
+        dirty_bounds: Option<&mut ffi::ARect>,
```

Comment:
I would avoid having a mutable reference like this if all possible. I would pass in a rectangle and then return a rectangle as a tuple with the lock.

Reply:
I instead considered putting it in the returned guard.

Reply:
Done that now, but it might make it less clear that when the user passes this to the function, they have to read back _the updated value_ from `NativeWindowBufferLockGuard`. But that would be the same when they already define their local variable `mut` for other reasons.